### PR TITLE
Introduce new parameters to allow vpnclient package generation for external radius based auth

### PR DESCRIPTION
**Pull request by ritwikbasu**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-network/2017-06-01/swagger/virtualNetworkGateway.json [hunk 1]
```diff
@@ -1350,11 +1350,31 @@
...
+          "enum": [
+            "EAPTLS",
+            "EAPMSCHAPv2"
+          ],
+          "x-ms-enum": {
+            "name": "AuthenticationMethod",
+            "modelAsString": true
+          }
+        },
+        "RadiusServerAuthCertificate": {
```
Comment (veronicagg) [inline arm-network/2017-06-01/swagger/virtualNetworkGateway.json]: property should be camel case

File: arm-network/2017-06-01/swagger/virtualNetworkGateway.json [hunk 2]
```diff
@@ -1350,11 +1350,31 @@
...
+          "type": "string",
+          "description": "The public certificate data for the radius server authentication certificate as a Base-64 encoded string. Required only if external radius authentication has been configured with EAPTLS authentication."
+        },
+        "ClientRootCertificates": {
```
Comment (veronicagg) [inline arm-network/2017-06-01/swagger/virtualNetworkGateway.json]: property name should be camel case

File: arm-network/2017-06-01/swagger/virtualNetworkGateway.json [hunk 3]
```diff
@@ -1350,11 +1350,31 @@
...
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "A list of client root certificates public certificate data encoded as Base-64 strings. Optional parameter for external radius based authentication with EAPTLS."
         }
       },
-      "required": [
```
Comment (veronicagg) [inline arm-network/2017-06-01/swagger/virtualNetworkGateway.json]: is removing this required property intentional in this PR?

**Review by veronicagg — CHANGES_REQUESTED** [review]
@ritwikbasu has this change been deployed to ARM already with the new API version?
BTW, you'll need to resolve conflicts with master, the initial commit of these specs has already have been made there.

File: arm-network/2017-06-01/swagger/virtualNetworkGateway.json (hunk 1 above)
Comment (ritwikbasu) [inline arm-network/2017-06-01/swagger/virtualNetworkGateway.json]: Fixed

File: arm-network/2017-06-01/swagger/virtualNetworkGateway.json (hunk 2 above)
Comment (ritwikbasu) [inline arm-network/2017-06-01/swagger/virtualNetworkGateway.json]: Fixed

**Comment by ritwikbasu** [comment]
Hi,

I have resolved the camel casing issues.

As for the other comments:


1.      has this change been deployed to ARM already with the new API version?

a.       The checkin has been made into NRP, and NRP already has the new API version. I am not sure that change has been rolled out yet.



2.      is removing this required property intentional in this PR?

a.       Yes – this is intentional, in preparation for the new API that we will expose.



3.      you'll need to resolve conflicts with master, the initial commit of these specs has already have been made there.

a.       When I checked on Friday, these changes were not there. That is why I pushed the files in to make my changes.

Please let me know if I am missing anything here.

File: arm-network/2017-06-01/swagger/virtualNetworkGateway.json (hunk 2 above)
Comment (veronicagg) [inline arm-network/2017-06-01/swagger/virtualNetworkGateway.json]: does the service reflect the same casing?

File: arm-network/2017-06-01/swagger/virtualNetworkGateway.json (hunk 1 above)
Comment (veronicagg) [inline arm-network/2017-06-01/swagger/virtualNetworkGateway.json]: does the service reflect the same casing?

**Comment by veronicagg** [comment]
@ritwikbasu thanks for the reply, I'll take another look once you publish the changes.
Regarding the changes being deployed to ARM or not, we should only merge once you know if the changes are deployed to ARM, the reason is that a spec committed in master would be used to generate SDKs that can be consumed by customers, therefore SDK calls need to work. Please do let me know, once you confirm that's the case. If you have other questions, feel free to let me know. Thanks!

**Comment by ritwikbasu** [comment]
Will create new pull request with all review comments

**Pull request closed without merging**